src/resolvers/git.cr

```diff
@@ -214,13 +218,26 @@ module Shards
       ref = git_ref(version)
 
       Dir.mkdir_p(install_path)
-      run "git --work-tree=#{Process.quote(install_path)} checkout #{Process.quote(ref.to_git_ref)} -- ."
+      temp_path = File.join(Dir.tempdir, "shards_#{Process.pid}")
+      Dir.mkdir_p(temp_path)
+      run "git --work-tree=#{Process.quote(temp_path)} checkout #{Process.quote(ref.to_git_ref)} -- ."
+      FileUtils.cp_r(File.join(temp_path, subdir, "."), install_path)
     end
 
     def commit_sha1_at(ref : GitRef)
       capture("git log -n 1 --pretty=%H #{Process.quote(ref.to_git_ref)}").strip
     end
 
+    def subdir
+      uri = parse_uri(git_url)
+      subdir = uri.query_params.fetch("subdir", "")
+      if subdir.blank? && git_url.includes?("?")
```

Comment:
This was specifically for the specs where it uses the file-based git repos. `URI` doesn't recognize the query params used there so some manual intervention is required.